specs/DefaultDownloadDialog.md

```diff
@@ -0,0 +1,221 @@
+                    else if (target_position == DownloadDialogPosition::kTopRight)
+                    {
+                        new_position = {x_from_right, padding};
+                    }
+                    else if (target_position == DownloadDialogPosition::kBottomLeft)
+                    {
+                        new_position = {padding, y_from_bottom};
+                    }
+                    else if (target_position == DownloadDialogPosition::kBottomRight)
+                    {
+                        new_position = {x_from_right, y_from_bottom};
+                    }
+                    CHECK_FAILURE(args->put_Position(new_position));
+                    return S_OK;
+                })
+                .Get(),
+            &m_downloadLayoutToken));
+    }
+}
+```
+
+```c#
+void ToggleDownloadDialogCmdExecuted(object target, ExecutedRoutedEventArgs e)
+{
+    webView.CoreWebView2.IsDefaultDownloadDialogVisible =
+        !webView.CoreWebView2.IsDefaultDownloadDialogVisible;
+}
+
+void DownloadDialogPositionCmdExecuted(object target, ExecutedRoutedEventArgs e)
+{
+    webView.CoreWebView2.DefaultDownloadDialogLayout += delegate (
+        object sender, CoreWebView2DefaultDownloadDialogLayoutEventArgs args)
+    {
+        int padding = 10;
+        int width = args.Width;
+        int height = args.Height;
+        System.Drawing.Rectangle bounds = args.LocalBounds;
+        string position = e.Parameter.ToString();
+        int x_from_right = bounds.Width - width - padding;
+        int y_from_bottom = bounds.Height - height - padding;
+        System.Drawing.Point new_position = args.Position;
+        if (position == "Top-left")
+        {
+            new_position = new System.Drawing.Point(padding, padding);
+        }
+        else if (position == "Top-right")
+        {
+            new_position = new System.Drawing.Point(x_from_right, padding);
+        }
+        else if (position == "Bottom-left")
+        {
+            new_position = new System.Drawing.Point(padding, y_from_bottom);
+        }
+        else if (position == "Bottom-right")
+        {
+            new_position = new System.Drawing.Point(x_from_right, y_from_bottom);
+        }
+        args.Position = new_position;
+    };
+}
+```
+
+# API Details
+
+
+```c#
+[uuid(9139c04d-8f37-42ae-8b63-01940c34d22f), object, pointer_default(unique)]
+interface ICoreWebView2_6 : ICoreWebView2_5
+{
+  /// `TRUE` if the default download dialog is currently visible.
+  [propget] HRESULT IsDefaultDownloadDialogVisible(
+      [out, retval] BOOL* isDefaultDownloadDialogVisible);
+
+  /// Sets the `IsDefaultDownloadDialogVisible` property. A download is
+  /// considered recent if it was downloaded in this instance of the webview.
+  /// If there has not yet been a recent download, setting the default download
+  /// dialog to visible displays all past downloads for the current profile.
+  /// Otherwise, the dialog will display only the recent downloads with a
```

Comment:
When "See more" is clicked, that will change the download dialog size, right? How does that impact the position? Does that trigger DefaultDownloadDialogLayout event?

Reply:
Yes, clicking "See more" changes the download dialog size and triggers a DefaultDownloadDialogLayout event. The event gives the opportunity to reposition the dialog considering the new height, which is provided in the event args